lua/terrortown/menus/gamemode/base_gamemodemenu.lua

```diff
@@ -177,3 +188,26 @@ end
 -- @hook
 -- @realm client
 function CLGAMEMODEMENU:Initialize() end
+
+---
+-- Populates the nav panel contents. The default is what is typically desired,
+-- however this extension point enables certain menus (like the Roles menu)
+-- to be populated in a more fitting way.
+-- @note This function can be overridden to create a custom nav panel.
+-- @param Panel navPanel The nav panel container
+-- @param Panel contentPanel The content panel container
+-- @hook
+-- @realm client
+function CLGAMEMODEMENU:FillNavPanel(navPanel, contentPanel, padding)
+    -- MAKE SEPARATE SUBMENULIST ON THE NAVAREA WITH A CONTENT AREA
+    local submenuList = vgui.Create("DSubmenuListTTT2", navPanel)
+    submenuList:Dock(FILL)
+    submenuList:SetPadding(padding)
+    submenuList:SetBasemenuClass(self, contentPanel)
+    if self.searchBarPlaceholder then
+        submenuList:SetSearchBarPlaceholder(self.searchBarPlaceholder)
+    end
+
+    -- REFRESH SIZE OF SUBMENULIST FOR CORRECT SUBMENU DEPENDENT SIZE
+    submenuList:InvalidateLayout(true)
+end
```

Comment:
I'm not sure I like it that GUI code is moved from the gui file to the class. The way the classes are designed, it was intended that they only contain data, no UI code. I'm not sure how complicated that change would be though. But if it is feasable, I'd like this part to be moved to the file where it was before.

The whole menu structure right now is defined in cl_help.lua. Down to the position of the search bar. Now this one nav panel is moved to a different file for no apparant reason. This makes mainting more tedious in my opinion.

Reply:
This is mostly an artifact of a different approach that wanted to let submenus define structurally *very* different UIs. That design didn't work out for what I wanted here, but I left the extension point mostly because I didn't see much reason to remove it again.

Reply:
thanks for reverting that change!